.github/workflows/main.yaml

```diff
@@ -140,50 +143,139 @@ jobs:
-          export WASMER_DIR=`pwd`/package
-          make test-integration
-        if: matrix.run_integration_tests && matrix.os != 'windows-latest'
+          ls target/release
+          alias wasmer=target/release/wasmer
+          ./target/release/wasmer compile --target=x86_64-unknown-linux-gnu tests/integration/cli/assets/qjs.wasm -o qjs_linux_from_mac.wjit
+          #./target/release/wasmer compile --target=x86_64-pc-windows-msvc tests/integration/cli/assets/qjs.wasm -o qjs_win_from_mac.wjit
+          mkdir cross
+          cp qjs_linux_from_mac.wjit cross
+          #cp qjs_win_from_mac.wjit cross
+          cp target/release/wasmer cross
+      #- name: Cross compile from Windows
+      #  if: matrix.os == 'windows-latest'
+      #  shell: bash
+      #  run: |
+      #    alias wasmer=target/release/wasmer
+      #    ./target/release/wasmer compile --target=x86_64-unknown-linux-gnu tests/integration/cli/assets/qjs.wasm -o qjs_linux_from_win.wjit
+      #    ./target/release/wasmer compile --target=x86_64-apple-darwin tests/integration/cli/assets/qjs.wasm -o qjs_mac_from_win.wjit
+      #    mkdir cross
+      #    cp qjs_linux_from_win.wjit cross
+      #    cp qjs_mac_from_win.wjit cross
+      #    cp target/release/wasmer.exe cross
+      - name: Upload Cross Compiled Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.cross_compilation_artifact_name }}
+          path: cross
       - name: Upload Artifacts
         uses: actions/upload-artifact@v2
         if: needs.setup.outputs.DOING_RELEASE == '1'
         with:
           name: ${{ matrix.artifact_name }}
           path: dist
 
-  release:
+  test-cross-compile-on-linux:
+    name: Test cross-compile on linux
+    needs: [setup, test]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Wasmer
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Run it
+        run: |
+          cp artifacts/cross_compiled_from_linux/wasmer .
+          chmod +x wasmer
+          ./wasmer run artifacts/cross_compiled_from_mac/qjs_linux_from_mac.wjit -- --eval "function greet(name) { return JSON.stringify('Hello, ' + name); }; print(greet('World'));"
+          #wasmer run artifacts/cross/cross_compiled_from_win/qjs_linux_from_win.wjit -- --eval "function greet(name) { return JSON.stringify('Hello, ' + name); }; print(greet('World'));"
+
+  test-cross-compile-on-mac:
+    name: Test cross-compile on macos
     needs: [setup, test]
+    runs-on: macos-latest
+    steps:
+      - name: Download Wasmer
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Run it
+        run: |
+          cp artifacts/cross_compiled_from_mac/wasmer .
+          chmod +x wasmer
+          ./wasmer run artifacts/cross_compiled_from_linux/qjs_mac_from_linux.wjit -- --eval "function greet(name) { return JSON.stringify('Hello, ' + name); }; print(greet('World'));"
```

Comment:
How are we testing that the result is what we expect?

Reply:
Good catch, currently we're not